pkg/common/telemetry/latency.go

```diff
@@ -0,0 +1,80 @@
+package telemetry
+
+import (
+	"sync"
+	"time"
+)
+
+// Latency is used to track timing between two specific events. It
+// is a generic version of CallCounter and can be used to measure latency between any two events.
+//
+// Example 1:
+//		func Foo() {
+//		    latency := StartLatencyMetric(metrics, "foo")
+//		 	call.AddLabel("food", "burgers")
+//  		// do something
+//	     	latency.Measure() // measure time elapsed between StartLatencyMetric() and Measure()
+//			// do other things
+//		}
+//
+// Example 2:
+//		func Bar() {
+//		    latency := StartLatencyMetric(metrics, "bar")
+//		 	call.AddLabel("food", "pizza")
+//  		// do something
+//	     	latency.MeasureAndReset() // emits metric for time elapsed between StartLatencyMetric() and Measure()
+//			// do other things
+//          latency.Measure() // emits metric for time elapsed between MeasureAndReset() and Measure()
+//		}
+//
+// Instances of this struct should only be created directly by this package
+// and its subpackages, which define the specific metrics that are emitted.
+// It is left exported for testing purposes.
```

Comment:
Since `StartLatencyMetric` returns `*Latency`, it would require the struct to be exported. Other option would be to return an exported interface implemented by this struct.